voice/src/main/kotlin/gateway/DefaultVoiceGatewayBuilder.kt

```diff
@@ -21,6 +21,7 @@ public class DefaultVoiceGatewayBuilder(
     public var client: HttpClient? = null
     public var reconnectRetry: Retry? = null
     public var eventFlow: MutableSharedFlow<VoiceEvent> = MutableSharedFlow(extraBufferCapacity = Int.MAX_VALUE)
+    public var isDeaf: Boolean = false
```

Comment:
`VoiceConnectionBuilder` already has the `selfDeaf` option, can't this be used instead?

Reply:
I suppose. I think the Voice Connection Builder could get cleaned up quite a bit though. 

Like I state in the PR description, you can set `selfDeaf` and a `streams` implementation.

Since `selfDeaf` is just a UI update right now it's not that much of an issue but if it gets updated to actually prevent voice receive it might be a good idea to set it depending on the streams implementation.

And maybe add some more functionality to it as well:

```kt
interface Streams {
	suspend fun setDeafen(value: Boolean)
	
	suspend fun setMute(ssrc: UInt, value: Boolean)
}
```

Reply:
I think this is a fair solution. Makes the deafening much behavior much more consistent, especially with this addition. Streams#setDeafen might not be needed, as it isn't provided for the UI deafen either. I could see the use for a VoiceConnection#(set)Deafen which would UI deafen and media sink deafen (if streams are open).